shared/src/commonMain/kotlin/org/hyperskill/app/manage_subscription/presentation/ManageSubscriptionReducer.kt

```diff
@@ -0,0 +1,60 @@
+package org.hyperskill.app.manage_subscription.presentation
+
+import org.hyperskill.app.manage_subscription.domain.analytics.ManageSubscriptionHyperskillAnalyticsEvent
+import org.hyperskill.app.manage_subscription.presentation.ManageSubscriptionFeature.Action
+import org.hyperskill.app.manage_subscription.presentation.ManageSubscriptionFeature.InternalAction
+import org.hyperskill.app.manage_subscription.presentation.ManageSubscriptionFeature.InternalMessage
+import org.hyperskill.app.manage_subscription.presentation.ManageSubscriptionFeature.Message
+import org.hyperskill.app.manage_subscription.presentation.ManageSubscriptionFeature.State
+import ru.nobird.app.presentation.redux.reducer.StateReducer
+
+private typealias ReducerResult = Pair<State, Set<Action>>
+
+class ManageSubscriptionReducer : StateReducer<State, Message, Action> {
+    override fun reduce(state: State, message: Message): ReducerResult =
+        when (message) {
+            Message.Initialize -> handleInitialize(state)
+            Message.ViewedEventMessage -> handleViewedEventMessage(state)
+            is InternalMessage.FetchSubscriptionSuccess -> handleFetchSubscriptionSuccess(message)
+            InternalMessage.FetchSubscriptionError -> handleFetchSubscriptionError()
+            Message.RetryContentLoading -> fetchSubscription(forceLoadFromNetwork = true)
+            Message.ManageSubscriptionClicked -> handleManageSubscriptionClicked(state)
+        }
+
+    private fun handleInitialize(state: State): ReducerResult =
+        if (state is State.Idle) {
+            fetchSubscription(forceLoadFromNetwork = false)
```

Comment:
In my opinion it's better to fetch subscription remotely, otherwise user can see outdated data when screen presented. So I suggest to remove forceLoadFromNetwork flag and always load from remote data source.